File: Source/Aladdin.CAPI.ANSI.PKCS11.Java/src/aladdin/capi/ansi/pkcs11/sign/ecdsa/VerifyData.java
package aladdin.capi.ansi.pkcs11.sign.ecdsa;
import aladdin.asn1.*;
import aladdin.asn1.ansi.x962.*;
import aladdin.capi.*; 
import aladdin.capi.pkcs11.*; 
import aladdin.capi.ansi.pkcs11.x962.*;
import java.security.*;
import java.io.*;

///////////////////////////////////////////////////////////////////////////
// Алгоритм проверки подписи DSA
///////////////////////////////////////////////////////////////////////////
public class VerifyData extends aladdin.capi.pkcs11.VerifyData
{
    // конструктор
	public VerifyData(Applet applet, long algID) 
    
        // сохранить переданные параметры
        { super(applet); this.algID = algID; } private final long algID;

	// параметры алгоритма
    @Override 
    protected Mechanism getParameters(Session sesssion, IParameters parameters)
	{
		// параметры алгоритма
		return new Mechanism(algID); 
	}
	// инициализировать алгоритм
    @Override
	public void init(IPublicKey publicKey, byte[] signature) 
        throws SignatureException, IOException
    {
        // преобразовать параметры алгоритма
        aladdin.capi.ansi.x962.IParameters parameters = 
            (aladdin.capi.ansi.x962.IParameters)publicKey.parameters(); 
        
        // раскодировать значение подписи
        ECDSASigValue encoded = new ECDSASigValue(Encodable.decode(signature)); 
        
        // вызвать базовую функцию
        super.init(publicKey, Encoding.encodeSignature(parameters, encoded)); 
    }
}

File: Source/Aladdin.CAPI.ANSI.PKCS11/Sign/ECDSA/VerifyData.cs
﻿using System;
using Aladdin.PKCS11; 

namespace Aladdin.CAPI.ANSI.PKCS11.Sign.ECDSA
{
    ///////////////////////////////////////////////////////////////////////////
    // Алгоритм проверки подписи DSA
    ///////////////////////////////////////////////////////////////////////////
    public class VerifyData : CAPI.PKCS11.VerifyData
    {
        // способ кодирования чисел
        private const Math.Endian Endian = Math.Endian.BigEndian; 
    
        // конструктор
	    public VerifyData(CAPI.PKCS11.Applet applet, ulong algID) 
    
            // сохранить переданные параметры
            : base(applet) { this.algID = algID; } private ulong algID;

	    // параметры алгоритма
        protected override Mechanism GetParameters(
            CAPI.PKCS11.Session sesssion, IParameters parameters)
	    {
		    // параметры алгоритма
		    return new Mechanism(algID); 
	    }
	    // инициализировать алгоритм
	    public override void Init(IPublicKey publicKey, byte[] signature)
        {
            // преобразовать параметры алгоритма
            ANSI.X962.IParameters parameters = (ANSI.X962.IParameters)publicKey.Parameters; 
        
            // раскодировать значение подписи
            ASN1.ANSI.X962.ECDSASigValue encoded = 
                new ASN1.ANSI.X962.ECDSASigValue(ASN1.Encodable.Decode(signature)); 

            // вызвать базовую функцию
            base.Init(publicKey, X962.Encoding.EncodeSignature(parameters, encoded)); 
        }
    }
}
